Sloth.Core/Services/CyberSourceBankReconcileService.cs

```diff
@@ -174,9 +195,10 @@ public class CyberSourceBankReconcileService : ICyberSourceBankReconcileService
 
                         var amount = decimal.Parse(paymentInfo.Amount);
 
-                        // create document number
-                        var documentNumber = await _context.GetNextDocumentNumber(tran.GetDbTransaction());
-                        var kfsTrackingNumber = await _context.GetNextKfsTrackingNumber(tran.GetDbTransaction());
+                        // ensure document and tracking numbers are created only once, in case this code gets
+                        // called multiple times due to a transient error
+                        documentNumber ??= await _context.GetNextDocumentNumber(tran.GetDbTransaction());
```

Comment:
CyberSource will never provide these values.

Reply:
But if a webhook request is sent multiple times due to a transient error. Which kfsTrackingNumber will be the correct one? This is just to ensure the same tracking number is used upon a retry.